### ISSUE TITLE
Separate the lifecycle of the kroxylicious tester from the test kroxylicious instance.

### ISSUE THREAD

**Issue by SamBarker** [kroxylicious/kroxylicious#526]
Sometimes we want the lifespan of the Kafka object (producer, consumer, admin) to extend beyond the lifespan of Kroxylicious instance.  We use this so that we can ensure that clients are capable of reconnecting through a replacement instance.

One way to solve this problem might be to break the association between the lifecycle of the tester the lifecycle of the kroxylicious.  That would retain the flexibility we need to write this sort of test:

```java

try(var tester = kroxyliciousTester();// No longer creates a kroxylicious instance
     var admin = tester.admin()) { 
     try(var kroxy = tester.kroxylicious()) { // Creates a Kroxylicious
        // ...
        //  do something with admin
     } 
     try(var kroxy = tester.kroxylicious()) { // Creates a Kroxylicious
        // ...
        // check admin reconnects
     }
}
```

It would also mean `tester.close()` could be given the semantics to ensure that everything it created has been closed, which is more obvious for the programmer to reason about.

_Originally posted by @k-wall in https://github.com/kroxylicious/kroxylicious/pull/519#discussion_r1285605037_

**Comment by robobario** [comment]
Does that conflict with this other issue https://github.com/kroxylicious/kroxylicious/issues/378, if we can make clients independent of the existence of a Kroxylicious instance? If we made the bootstrap address something obtained dynamically from a running Kroxylicious it wouldn't work.

Another way could be to keep the same API and make KroxyliciousTester able to clean up dangling clients on close, and in special cases like the resilience test we construct them in the test using config supplied by the tester so they are detached from Tester.

**Comment by k-wall** [comment]
> ```java
> kroxyliciousTester
> ```



> Does that conflict with this other issue #378, if we can make clients independent of the existence of a Kroxylicious instance? If we made the bootstrap address something obtained dynamically from a running Kroxylicious it wouldn't work.

That's true.  At the moment the config is providing the bootstrap address, so providing the tester had sight of the configuration it would be able to create the client.   However, this would fail if ever we change Kroxy to be capable of binding dynamic ports.

The other alternative is to delay the construction of the client until kroxy has been brought up for the first time.

**Comment by gracegrimwood** [comment]
Triaged 16/08/23 - needs further discussion
